fractal_graphing/algos/algos.go

```diff
@@ -0,0 +1,98 @@
+// Copyright © 2016 Pawel Rozlach.
+// License: https://creativecommons.org/licenses/by-nc-sa/4.0/
+
+// Based on work by Alan A. A. Donovan & Brian W. Kernighan
+// which can be found at:
+// https://github.com/adonovan/gopl.io.git
+
+// Mandelbrot emits a PNG image of the Mandelbrot fractal.
+package algos
+
+import "math/cmplx"
+
+//(sur) Is it a problem that I make func type private and at the same time
+// map using it public ?
+type AlgoFunc func(r, i float64) (uint8, uint8)
+
+// (sur) Is there a way to do _getattr_ in a simple way ? Or is the "map"
+// approach idiomatic enough ?
+var Str2funcMapping = map[string]AlgoFunc{
+	"newton":         Newton,
+	"acos":           Acos,
+	"mandelbrotC64":  MandelbrotC64,
+	"mandelbrotC128": MandelbrotC128,
+	"sqrt":           Sqrt,
+}
+
+// Mandelbrot algo, but with complex128 type resolution
+func MandelbrotC128(r, i float64) (uint8, uint8) {
+	var z complex128 = complex(r, i)
+	const iterations = 20000
+
+	var v complex128
+	for n := 0; n < iterations; n++ {
+		v = v*v + z
+		if cmplx.Abs(v) > 2 {
+			blue := uint8(real(v)*128) + 127
+			red := uint8(imag(v)*128) + 127
+			return blue, red
+		}
+	}
+	return 0, 0
+}
+
+// Mandelbrot algo, but with complex64 type resolution
+func MandelbrotC64(r, i float64) (uint8, uint8) {
+	var z complex64 = complex(float32(r), float32(i))
+	const iterations = 20000
+
+	var v complex64
+	for n := 0; n < iterations; n++ {
+		v = v*v + z
+		//(sur) I have not found a better way, golang seems to support only
+		// float64 arithmetics :/
```

Comment:
Yeah, I am also in the dark here. I have made some research but nothing definitive - seems like most of stdlib stuff supports only float64 arithmetics :/